rabbit-escape-engine/src/rabbitescape/engine/behaviours/Drowning.java

```diff
@@ -0,0 +1,67 @@
+package rabbitescape.engine.behaviours;
+
+import static rabbitescape.engine.CellularDirection.DOWN;
+import static rabbitescape.engine.CellularDirection.UP;
+
+import rabbitescape.engine.Behaviour;
+import rabbitescape.engine.BehaviourTools;
+import rabbitescape.engine.CellularDirection;
+import rabbitescape.engine.ChangeDescription.State;
+import rabbitescape.engine.Rabbit;
+import rabbitescape.engine.WaterRegion;
+import rabbitescape.engine.World;
+
+public class Drowning extends Behaviour
+{
+    @Override
+    public void cancel()
+    {
+    }
+
+    @Override
+    public boolean checkTriggered( Rabbit rabbit, World world )
+    {
+        int yCoordinate = rabbit.y;
+        CellularDirection directionToCheck = UP;
+        if ( rabbit.onSlope )
+        {
+            // The rabbit's head is at the bottom of the cell above.
+            yCoordinate = rabbit.y - 1;
+            directionToCheck = DOWN;
+        }
+        // TODO Find out why the rabbit's y coordinate is allowed to be larger than the size of the world (see solution for easy-12).
```

Comment:
This is kind of TODONE, but I've left this here as a reminder to find out the limits on x and y (i.e. what should this bounds check say?)